File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        leftPointer = 0
        for rightPointer in range(len(nums)):
            if nums[rightPointer] != 0:
                nums[leftPointer], nums[rightPointer] = nums[rightPointer], nums[leftPointer]
                leftPointer += 1

File: 0283-move-zeroes/0283-move-zeroes.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function moveZeroes(nums: number[]): void {
    let leftPointer = 0;
    for (let rightPointer = 0; rightPointer < nums.length; rightPointer++) {
        if (nums[rightPointer] !== 0) {
            [nums[leftPointer], nums[rightPointer]] = [nums[rightPointer], nums[leftPointer]]
            leftPointer += 1
        }
    }
};